### PR TITLE
Feature/100 policy setup

### PR DESCRIPTION
**Pull request by Izzzu**
This PR adds a simple set up of policies.

Based on spike agera-edc/MinimumViableDataspaceFork#99 

This PR contains following changes:

- As did mock PR was already merged this PR uses did mock instead of iam mock
- Terraform files modified to be able to specify region for a connector
- Postman collection updated with another asset, policy that restricts asst 2 to EU region and contract definition for added policy and asset
- SeedPoliciesExtension added to bind policies to the rule binding registry (according to the doc: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/policy-engine.md)
- System Test CatalogClientTest modified to test usage of the newly added policies
    - test 1: a call to a catalog of consumer with EU region returns 2 assets
    - test 2: a call to a catalog of consumer with US region returns 1 asset 
- env variables 
EDC_CATALOG_CACHE_EXECUTION_DELAY_SECONDS and EDC_CATALOG_CACHE_EXECUTION_PERIOD_SECONDS changed to value 5sec, because of a bug [#1366](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1366)

### CODE REVIEW

File: extensions/policies/src/main/java/org/eclipse/dataspaceconnector/mvd/SeedPoliciesExtension.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.eclipse.dataspaceconnector.spi.contract.offer.ContractDefinitionService;
+import org.eclipse.dataspaceconnector.spi.policy.RuleBindingRegistry;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import static org.eclipse.dataspaceconnector.ids.spi.policy.IdsPolicyExpressions.ABS_SPATIAL_POSITION;
+
+public class SeedPoliciesExtension implements ServiceExtension {
```
Comment (algattik) [inline extensions/policies/src/main/java/org/eclipse/dataspaceconnector/mvd/SeedPoliciesExtension.java:16]: javadoc for class and/or method would be more than welcome

**Review by algattik — APPROVED** [review]

File: deployment/data/MVD.postman_collection.json [hunk 2]
```diff
@@ -124,8 +207,49 @@
...
+								"type": "text/javascript"
+							}
+						}
+					],
+					"request": {
+						"method": "POST",
+						"header": [],
+						"body": {
+							"mode": "raw",
+							"raw": "{\n    \"id\": \"8dfd8f18-09ad-41f9-a2d8-368889d3e36e\",\n    \"accessPolicyId\": \"6a99c1bb-74ad-41a7-b73a-93233ffdbfb8\",\n    \"contractPolicyId\": \"6a99c1bb-74ad-41a7-b73a-93233ffdbfb8\",\n        \"criteria\": [\n            {\n                \"left\": \"asset:prop:id\",\n                \"op\": \"=\",\n                \"right\": \"test-document-2\"\n            }\n        ]\n}",
```
Comment (cpeeyush) [inline deployment/data/MVD.postman_collection.json:232]: Seems like transfers will fail for this contract definition as we do not have an actual blob named as `text-document-2.txt` in backend storage account.

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -80,16 +80,22 @@ jobs:
     outputs:
       company1_edc_host: ${{ steps.runterraform.outputs.company1_edc_host }}
       company2_edc_host: ${{ steps.runterraform.outputs.company2_edc_host }}
+      company3_edc_host: ${{ steps.runterraform.outputs.company3_edc_host }}
       company1_key_vault: ${{ steps.runterraform.outputs.company1_key_vault }}
       company2_key_vault: ${{ steps.runterraform.outputs.company2_key_vault }}
+      company3_key_vault: ${{ steps.runterraform.outputs.company3_key_vault }}
       company1_api_key: ${{ steps.runterraform.outputs.company1_api_key }}
       company2_api_key: ${{ steps.runterraform.outputs.company2_api_key }}
+      company3_api_key: ${{ steps.runterraform.outputs.company2_api_key }}
```
Comment (cpeeyush) [inline .github/workflows/deploy.yaml]:
```suggestion
      company3_api_key: ${{ steps.runterraform.outputs.company3_api_key }}
```

File: extensions/policies/src/main/java/org/eclipse/dataspaceconnector/mvd/SeedPoliciesExtension.java (hunk 1 above)
Comment (Izzzu) [reply extensions/policies/src/main/java/org/eclipse/dataspaceconnector/mvd/SeedPoliciesExtension.java:16]: added

File: deployment/data/MVD.postman_collection.json (hunk 2 above)
Comment (Izzzu) [reply deployment/data/MVD.postman_collection.json:232]: I added a blob named text-document-2.txt to the terraform. Thanks

Comment (Izzzu) [reply deployment/data/MVD.postman_collection.json:232]: Here's the transfer tested for tet-document-2.txt 
http://mvd589-company2-mvd.northeurope.azurecontainer.io/transfer-history

**Review by cpeeyush — APPROVED** [review]

**Pull request merged**